unittests/comprehensive/test_ForceDependentSlip.cpp

```diff
@@ -0,0 +1,280 @@
+  shapeNode->getDynamicsAspect()->setSecondarySlipCompliance(0);
+
+  // Put the body into position
+  Eigen::Isometry3d tf = Eigen::Isometry3d::Identity();
+  tf.translation() = Eigen::Vector3d(0.0, 0.0, -floorHeight / 2.0);
+  body->getParentJoint()->setTransformFromParentBodyNode(tf);
+
+  return floor;
+}
+
+//==============================================================================
+SkeletonPtr createCylinder(
+    double _radius,
+    double _height,
+    const Eigen::Vector3d& _position = Eigen::Vector3d::Zero(),
+    const Eigen::Vector3d& _orientation = Eigen::Vector3d::Zero())
+{
+  SkeletonPtr cylinder = createObject(_position, _orientation);
+
+  BodyNode* bn = cylinder->getBodyNode(0);
+  std::shared_ptr<Shape> cylinderShape(new CylinderShape(_radius, _height));
+  bn->createShapeNodeWith<VisualAspect, CollisionAspect, DynamicsAspect>(
+      cylinderShape);
+
+  return cylinder;
+}
+
+//==============================================================================
+TEST(ForceDependentSlip, BoxSlipVelocity)
+{
+  using Eigen::Vector3d;
+  const double mass = 5.0;
+  const double slip = 0.02;
+  auto skeleton1 = createBox({0.3, 0.3, 0.3}, {-0.5, 0, 0.15});
+  skeleton1->setName("Skeleton1");
+  auto skeleton2 = createBox({0.3, 0.3, 0.3}, {+0.5, 0, 0.15});
+  skeleton2->setName("Skeleton2");
+
+  auto body1 = skeleton1->getRootBodyNode();
+  body1->setMass(mass);
+  auto body1Dynamics = body1->getShapeNode(0)->getDynamicsAspect();
+
+  EXPECT_DOUBLE_EQ(body1Dynamics->getFrictionCoeff(), 1.0);
+
+  body1Dynamics->setSlipCompliance(slip);
+  body1Dynamics->setFirstFrictionDirection(Vector3d::UnitX());
+  EXPECT_DOUBLE_EQ(body1Dynamics->getSlipCompliance(), slip);
+  EXPECT_EQ(body1Dynamics->getFirstFrictionDirection(), Vector3d::UnitX());
+
+  auto body2 = skeleton2->getRootBodyNode();
+  body2->setMass(mass);
+  auto body2Dynamics = body2->getShapeNode(0)->getDynamicsAspect();
+  body2Dynamics->setFirstFrictionDirection(Vector3d::UnitX());
+  EXPECT_EQ(body2Dynamics->getFirstFrictionDirection(), Vector3d::UnitX());
+  EXPECT_DOUBLE_EQ(body2Dynamics->getFrictionCoeff(), 1.0);
+
+  // Create a world and add the rigid bodies
+  auto world = createWorld();
+  world->addSkeleton(createFloor());
+  world->addSkeleton(skeleton1);
+  world->addSkeleton(skeleton2);
+
+  const auto numSteps = 2000;
+  const double extForce = 10.0;
+  for (auto i = 0u; i < numSteps; ++i)
+  {
+    body1->addExtForce({extForce, 0, 0});
+    body2->addExtForce({extForce, 0, 0});
+    world->step();
+
+    if (i > 1000)
+    {
+      // The velocity of body1 should stabilize at F_ext * slip = 0.2 m/s
+      EXPECT_NEAR(extForce * slip, body1->getLinearVelocity().x(), 1e-4);
+      EXPECT_NEAR(0.0, body1->getLinearVelocity().y(), 1e-4);
+
+      // The second box should remain at rest because of friction
+      EXPECT_NEAR(0.0, body2->getLinearVelocity().x(), 1e-4);
+      EXPECT_NEAR(0.0, body2->getLinearVelocity().y(), 1e-4);
```

Comment:
are the two boxes going to bump into each other in this test? body2 starts at `0.5`, while body1 starts at `-0.5` and they both have positive force applied. it seems like they are destined to bump into each other. perhaps apply a negative force instead to avoid that?

Reply:
maybe they won't due to the time duration, but it seems like it would be safer to push in the opposite direction or swap their parameters

Reply:
I set them farther apart in their y coordinates.  https://github.com/azeey/dart/pull/9/commits/334a9375a6c58667e28069b089c96db51c90c9b6